kernel/src/arch/x86/smp/smp.cpp

```diff
@@ -177,7 +177,171 @@ namespace smp
+                                                        /*Numlock and Scroll Lock:*/ 0,
+                                                        0,
+                                                        '7',
+                                                        '8',
+                                                        '9',
+                                                        '-',
+                                                        '4',
+                                                        '5',
+                                                        '6',
+                                                        '+',
+                                                        '1',
+                                                        '2',
+                                                        '3',
+                                                        '0',
+                                                        '.',
+                                                        /* three gaps */ 0,
+                                                        0,
+                                                        0,
+                                                        /* F11, F12 */ 0,
+                                                        0};
+                        // 0x1B = ESCAPE; 0x08 = BACKSPACE; 0x09 = HORIZONAL TAB, used to represent TAB
+                        // 0x0D = CARRIAGE RETURN, used to represent ENTER
+                        // 0x11 = DEVICE CONTROL 1, used to represent LEFT CONTROL; 0x0E = SHIFT OUT, used to represent LEFT SHIFT
+                        // 0x0F = SHIFT IN, used to represent RIGHT SHIFT; 0x13 = DEVICE CONTROL 3, used to represent LEFT ALT
+                        // 0x02 = START OF TEXT, used to represent CAPS LOCK
+                        // TODO: Handle F1-F12, Numlock and Scroll Lock, multimedia keys
+                        // also Home, Page Up, etc
+                        // TODO eventually: distinguish Keypad / (0xE0, 0x35) and enter (0xE0, 0x1C) from normal, and RIGHT CTRL and ALT from LEFT
+                        char decoded_keypress = 0;
+                        bool is_pressed = true; // false means released
+
+                        if (encoded_keypress < (sizeof(us_qwerty_translation) / sizeof(us_qwerty_translation[0])))
+                        {
+                            decoded_keypress = us_qwerty_translation[encoded_keypress];
+                            is_pressed = true;
+                        }
+                        else if (encoded_keypress - 0x80 >= 0 &&
+                                 encoded_keypress - 0x80 < (sizeof(us_qwerty_translation) / sizeof(us_qwerty_translation[0])))
+                        {
+                            decoded_keypress = us_qwerty_translation[encoded_keypress - 0x80];
+                            is_pressed = false;
+                        }
+                        if (decoded_keypress != 0)
```

Comment:
Refactor debug prints into another function. Also avoid `strcpy`s, just modify where stuff points to